tests/rkt_tests.go

```diff
@@ -118,21 +118,36 @@ func spawnOrFail(t *testing.T, cmd string) *gexpect.ExpectSubprocess {
 	return child
 }
 
-func waitOrFail(t *testing.T, child *gexpect.ExpectSubprocess, shouldSucceed bool) {
+func getExitStatus(err error) int {
+	if err == nil {
+		return 0
+	}
+	if exiterr, ok := err.(*exec.ExitError); ok {
+		// the program has exited with an exit code != 0
+		if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+			return status.ExitStatus()
+		}
+	}
+	return -1
+}
+
+func waitOrFail(t *testing.T, child *gexpect.ExpectSubprocess, expectedStatus int) {
 	err := child.Wait()
+	status := getExitStatus(err)
+	if status != expectedStatus {
+		t.Fatalf("rkt terminated with unexpected status %d, expected %d\nOutput:\n%s", status, expectedStatus, child.Collect())
+	}
 	switch {
-	case !shouldSucceed && err == nil:
+	case expectedStatus != 0 && err == nil:
```

Comment:
For this to be executed, `expectedStatus` has to be the same as `status` from `getExitStatus`. This will happen only when `expectedStatus` is `0` and `err` is `nil`, or when `expectedStatus` not `0` and is equal to the result of `getExitStatus`, but that can happen only when `err` is not `nil`. So both `case expectedStatus != 0 && err == nil` and `expectedStatus == 0 && err != nil` are impossible at this point.

In short - I think you can just remove this `switch` statement.